### ISSUE TITLE
infra: verify GitHub pull requests will full power of our CI

### ISSUE THREAD

**Issue by Totktonada** [tarantool/tarantool#5258]
Tarantool version: 2.6.0-39-ga10212377.

We have large build matrix (different Linux distros, Mac OS, FreeBSD, various special builds), which is implemented using GitLab CI (we're [moving](https://github.com/tarantool/tarantool/issues/4894) to it from Travis CI, because only [5 parallel jobs](https://github.com/tarantool/tarantool/issues/4156#issuecomment-483435050) is not enough for us). The problem is that GitLab CI [does not run](https://gitlab.com/gitlab-org/gitlab/-/issues/5667) for pull requests from external repositories. It is especially important, since we [want](https://github.com/tarantool/doc/issues/1443) to suggest external contributors to send us GitHub PRs (instead of emails).

We should run full testing for pull requests to simplify possible problems catching for us and contributors.

Maybe even consider moving to GitHub Actions. It is modern, popular and evolves fast. It supports using our own machines. The only downside I know is that it stores build logs only for 3 months. This task requires considerable amount of work, though.

**Comment by tsafin** [comment]
FWIW, here is my attempt to enable cmake build via Github Actions https://github.com/tsafin/tarantool/actions/runs/227281968

**Comment by Totktonada** [comment]
Some jobs were moved to GitHub Actions in the scope of #5294. Remaining jobs and problems are tracked in #5644, #5660, #5661, #5662, #5663. In fact, I'll be satisfied with pull request testing after #5662, where packages will be built against different Linux distributions. Since it is tracked now in another issue, there is nothing to track here.
